### ISSUE TITLE
Exponent notation is not parsed correctly in transform attribute

### ISSUE THREAD

**Issue by OzymandiasTheGreat** [chrvadala/transformation-matrix#83]
So this took me couple hours to debug. When parsing transform attribute and using `toSVG` on it, it's not uncommon to get values in exponent notation. Upon parsing these values again, exponent part is ignored and we get a very different matrix.
You should either update the parser or if that's not viable, you could check for exponent and apply some rounding.
I don't think dropping decimals that far down would affect results in any way.

**Comment by chrvadala** [comment]
Hi @OzymandiasTheGreat,
I looked at the documentation and I can confirm that a transformation matrix data types is `matrix(<number>, <number>, <number>, <number>, <number>, <number>)`.


`<number>` data type can be one of the following https://developer.mozilla.org/en-US/docs/Web/CSS/number#Valid_numbers, so scientific notation is in the list of accepted representations.

````
12          A raw <integer> is also a <number>.
4.01        Positive fraction
-456.8      Negative fraction
0.0         Zero
+0.0        Zero, with a leading +
-0.0        Zero, with a leading -
.60         Fractional number without a leading zero
10e3        Scientific notation
-3.4e-2     Complicated scientific notation
````

Can you provide an example that reproduces the bug?

**Comment by OzymandiasTheGreat** [comment]
Of course, here's what I'm working with:
```javascript
var m = require("transformation-matrix")
const matrix = m.compose(m.fromDefinition(m.fromTransformAttribute("rotate(90, 395, 45)")))
console.log(matrix)
// {a: 6.123233995736766e-17, b: 1, c: -1, d: 6.123233995736766e-17, e: 440, f: -350}
console.log(m.compose(m.fromDefinition(m.fromTransformAttribute(m.toSVG(matrix)))))
// {a: 6.123233995736766, b: 1, c: -1, d: 6.123233995736766, e: 440, f: -350}
```
Everything starting with `e` is dropped. `toSVG` does produce correct matrix attribute, so the issue is with the parser.

For now I'm using `smoothMatrix` before `toSVG` and everything works as expected, but I doubt that's a workable solution at scale.

**Comment by chrvadala** [comment]
This is definitely a bug with `fromTransformAttribute` parser that it isn't able to parse exponential values.  I have to look at this descriptor https://github.com/chrvadala/transformation-matrix/blob/master/src/fromTransformAttribute.pegjs but, as per documentation, this parser is still in beta due to the beta pegjs dependency.

If you want to parse just matrixes that are in the form `matrix(a,b,c,d,e,f)` I suggested you the more reliable function `fromString`. I wrote some tests on the linked PR and I can confirm that `fromString` works as expected.
